--- a/build/resources/0ad.sh    2026-01-18 23:38:28.693460280 -0500
+++ b/build/resources/0ad.sh    2026-01-18 23:36:32.725715459 -0500
@@ -1,9 +1,12 @@
 #!/bin/sh
 
-pyrogenesis=$(which pyrogenesis 2>/dev/null)
-if [ -x "$pyrogenesis" ]; then
-	"$pyrogenesis" "$@"
-else
-	echo "Error: pyrogenesis not found in ($PATH)"
-	exit 1
-fi
+pyrogenesis=$(command -v pyrogenesis) || { echo "pyrogenesis not found"; exit 1; }
+gamemoderun=$(command -v gamemoderun)
+mangohud=$(command -v mangohud)
+
+cmd=("$pyrogenesis")
+
+[ -x "$mangohud" ] && cmd=("$mangohud" "${cmd[@]}")
+[ -x "$gamemoderun" ] && cmd=("$gamemoderun" "${cmd[@]}")
+
+exec "${cmd[@]}" "$@"
